packages/coverage/src/coverage.ts

```diff
@@ -0,0 +1,269 @@
+import path from 'path';
+import fs from 'fs';
+import {
+  BettererDiff,
+  BettererRun,
+  BettererSerialiser,
+  BettererTest,
+  BettererTestOptionsComplex
+} from '@betterer/betterer';
+import { BettererConstraintResult } from '@betterer/constraints';
+import { BettererLog, BettererLogs } from '@betterer/logger';
+import {
+  BettererCoverageDiff,
+  BettererCoverageIssue,
+  BettererCoverageIssues,
+  BettererCoverageTest,
+  BettererCoverageTestOptions,
+  coverageAttributes,
+  IstanbulCoverageSummary,
+  IstanbulCoverageSummaryType,
+  IstanbulFileCoverage
+} from './types';
+import { reduceJoinRecords, uncoveredFile } from './helpers';
+import { isRight } from 'fp-ts/Either';
+import { BettererError } from '@betterer/errors';
+import { Validation } from 'io-ts';
+
+class BettererCoverageTestΩ
+  implements BettererTestOptionsComplex<BettererCoverageIssues, BettererCoverageIssues, BettererCoverageDiff>
+{
+  private readonly excludes: Array<RegExp>;
+  private readonly includes: Array<RegExp>;
+  private readonly coverageReportPath: string;
+  private readonly baseDir: string;
+  private readonly onlyTotalCoverage: boolean;
+
+  constructor(options: BettererCoverageTestOptions) {
+    this.baseDir = path.resolve(options.baseDir);
+    this.coverageReportPath = path.resolve(this.baseDir, options.coverageFile);
+    this.excludes = options.excludeFiles;
```

Comment:
Is it possible to make the `includes` and `excludes` line up more with the API of the `BettererFileTest`?

Reply:
I thought about it, but a) it would require some more shenanigans with the return type of coverage() (The BettererFileTestBase interface is too restrictive to just use it in this case) and b) I find it confusing that one of the functions takes a glob, but the other takes a regex, so I don't really want to implement it this way.